src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -365,6 +371,14 @@ private static IEnumerable<SnapshotSpan> Difference<T>(IEnumerable<ITagSpan<T>>
                     latestEnumerator.Dispose();
                     previousEnumerator.Dispose();
                 }
+
+                return new DiffResult(added, removed);
+            }
+
+            private static void AddToList<T>(ref List<T> list, T item)
+            {
+                list = list ?? new List<T>();
+                list.Add(item);
```

Comment:
Because it's a micro optimization that in general doesn't seem like it will do anything.

--
Also, a better optimization would be to pool.  These lists may be very large.  And they're going to be created, and then thrown away immediately when the DiffResult is created.  If we pool them, then we only ever have as many lists created as needed to service the max number of threads doing diffing simultaneously.